lib/workflow/stores/mongo.js

```diff
@@ -15,6 +15,17 @@ mongoStoreFactory.$inject = [
 function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _) {
     var exports = {};
 
+    exports.publishRecordByGraphId = function (graphId, event) {
+        return waterline.graphobjects.needOne({ instanceId: graphId })
```

Comment:
Actually, couldn't you just pass the graph object you receive from the `findAndModifyMongo` queries below into this method? Rather than doing this call.